core/src/actions/session.ts

```diff
@@ -25,16 +27,42 @@ export class SessionCreate extends CLSAction {
     params,
   }: {
     connection: Connection;
-    params: { [key: string]: string };
+    params: { email: string; password?: string; requestId?: string };
   }) {
+    if (!params.password && !params.requestId) {
+      throw new Errors.AuthenticationError(
+        `password or an oAuth requestId is required`
+      );
+    }
+
     const teamMember = await TeamMember.findOne({
       where: { email: params.email.toLocaleLowerCase() },
     });
-    if (!teamMember)
+    if (!teamMember) {
       throw new Errors.AuthenticationError("team member not found");
+    }
 
-    const match = await teamMember.checkPassword(params.password);
-    if (!match) throw new Errors.AuthenticationError("password does not match");
+    if (params.password) {
+      const match = await teamMember.checkPassword(params.password);
+      if (!match)
+        throw new Errors.AuthenticationError("password does not match");
+    } else {
+      const oauthRequest = await OAuthRequest.findOne({
+        where: { id: params.requestId, consumed: false },
```

Comment:
It feels a little odd that the identities are still returned to the user for a consumed oauth request (going to /session/sign-in?requestId=req_ABCDE), but it's probably ok

Reply:
That's a good idea! https://github.com/grouparoo/grouparoo/pull/2667/commits/857d6871bec0488089451b7920e3857d2298976e prevents showing identities for consumed oAuth requests